src/Transmogrification.cpp

```diff
@@ -553,15 +547,120 @@ bool Transmogrification::SuitableForTransmogrification(Player* player, ItemTempl
     if (!IgnoreReqLevel && player->getLevel() < proto->RequiredLevel)
         return false;
 
+    return true;
+}
+
+bool Transmogrification::SuitableForTransmogrification(ObjectGuid guid, ItemTemplate const* proto) const
+{
+    if (!guid || !proto)
+        return false;
+
+    if (proto->Class != ITEM_CLASS_ARMOR &&
+        proto->Class != ITEM_CLASS_WEAPON)
+        return false;
+
+    // Skip all checks for allowed items
+    if (IsAllowed(proto->ItemId))
+        return true;
+
+    if (!CheckPureProtoRequirements(proto))
+        return false;
+
+    auto playerGuid = guid.GetCounter();
+    CharacterCacheEntry const* playerData = sCharacterCache->GetCharacterCacheByGuid(guid);
+    if (!playerData)
+        return false;
+
+    uint8 playerRace = playerData->Race;
+    uint8 playerLevel = playerData->Level;
+    uint32 playerRaceMask = 1 << (playerRace - 1);
+    uint32 playerClassMask = 1 << (playerData->Class - 1);
+    TeamId playerTeamId = Player::TeamIdForRace(playerRace);
+
+    std::unordered_map<uint32, uint32> playerSkillValues;
+    if (QueryResult resultSkills = CharacterDatabase.Query("SELECT `skill`, `value` FROM `character_skills` WHERE `guid` = {}", playerGuid))
+    {
+        do
+        {
+            Field* fields = resultSkills->Fetch();
+            uint16 skill = fields[0].Get<uint16>();
+            uint16 value = fields[1].Get<uint16>();
+            playerSkillValues[skill] = value;
+        } while (resultSkills->NextRow());
+    }
+    else {
+        LOG_ERROR("module", "Transmogification could not find skills for player with guid {} in database.", playerGuid);
+        return false;
+    }
+
+    if ((proto->Flags2 & ITEM_FLAGS_EXTRA_HORDE_ONLY) && playerTeamId != TEAM_HORDE)
+        return false;
+
+    if ((proto->Flags2 & ITEM_FLAGS_EXTRA_ALLIANCE_ONLY) && playerTeamId != TEAM_ALLIANCE)
+        return false;
+
+    if (!IgnoreReqClass && (proto->AllowableClass & playerClassMask) == 0)
+        return false;
+
+    if (!IgnoreReqRace && (proto->AllowableRace & playerRaceMask) == 0)
+        return false;
+
+    if (!IgnoreReqLevel && playerLevel < proto->RequiredLevel)
+        return false;
+
+    if (!IgnoreReqSkill && proto->RequiredSkill != 0)
+    {
+        if (playerSkillValues[proto->RequiredSkill] == 0)
+            return false;
+        else if (playerSkillValues[proto->RequiredSkill] < proto->RequiredSkillRank)
+            return false;
+    }
+
+    if (proto->SubClass > 0 && playerSkillValues[proto->GetSkill()] == 0)
+    {
+        if (proto->Class == ITEM_CLASS_ARMOR)
+        {
+            if (!AllowMixedArmorTypes)
+                return false;
+        }
+        else if (proto->Class == ITEM_CLASS_WEAPON)
+        {
+            if (!AllowMixedWeaponTypes)
+                return false;
+        }
+        else
+            return false;
```

Comment:
you can:
- merge the first two ifs into one
- convert else if into if and merge again the two conditions
- remove the else and leave return false

Reply:
Yeah, that's actually the old code just moved but true. Could take the chance to clean it all up a bit. I'll have a look at the 3 methods checking if the item is suitable for a player to transmog :)

Reply:
> you can:
> 
> * merge the first two ifs into one
> * convert else if into if and merge again the two conditions
> * remove the else and leave return false

Been thinking about how to simplify this a bit but the `else if` and `else` actually have their place here. Even wrapped if's have a sense coming up with the `else`.

More readable this would be:
```cpp

// Check for Item being an armor
if (proto->Class == ITEM_CLASS_ARMOR && !AllowMixedArmorTypes)
{
    return false;
}
        
// Check for Item being a weapon
if (proto->Class == ITEM_CLASS_WEAPON && !AllowMixedWeaponTypes)
{
    return false;
}

// The else
if (proto->Class != ITEM_CLASS_WEAPON && proto->Class != ITEM_CLASS_ARMOR)
{
    return false;
}
```

So in logical terms:
```
( a ∧ ¬ b ) ∨ ( c ∧ ¬ d ) ∨ ( ¬ a ∧ ¬ c )
```

Cannot really shorten it, just make it more readable. But that would technically be less efficient. Even if nobody would ever notice that on that level.

Reply:
I think your example text is perfect without the // `the else` comment 😁

Reply:
I didn't have the time to finish it last night sadly but I prepared a commit to clean up all those checks where I can :)
Coming tomorrow at the latest